src/firewall/core/fw.py

```diff
@@ -1090,7 +1090,20 @@ def flush(self, use_transaction=None):
         if use_transaction is None:
             transaction.execute(True)
 
-    def set_policy(self, policy, use_transaction=None):
+    def _set_policy_build_rules(self, backend, policy, drop_details=None):
+        assert policy in ("ACCEPT", "DROP", "PANIC")
+        if policy == "DROP":
+            if drop_details is None:
+                drop_details = {
+                    "INPUT": "DROP",
+                    "OUTPUT": "DROP",
+                    "FORWARD": "DROP",
+                }
+        else:
+            drop_details = None
```

Comment:
At this point, lets guarantee that `drop_details` is valid. That allows simpler logic down the call chain.

Reply:
The cognitive load argument is only because of the variable name. Pick a better name.